validator_client/src/block_service.rs

```diff
@@ -210,14 +219,19 @@ impl<T: SlotClock + 'static, E: EthSpec> BlockService<T, E> {
             .randao_reveal(&validator_pubkey, slot.epoch(E::slots_per_epoch()))
             .ok_or_else(|| "Unable to produce randao reveal".to_string())?;
 
-        let block = self
+        let mut block = self
             .beacon_node
             .http
             .validator()
-            .produce_block(slot, randao_reveal)
+            .produce_block(slot, randao_reveal, self.graffiti)
             .await
             .map_err(|e| format!("Error from beacon node when producing block: {:?}", e))?;
 
+        // If graffiti has been set in the validator client, overwrite any graffiti sent from the beacon node.
+        if let Some(graffiti) = self.graffiti {
+            block.body.graffiti = graffiti;
+        }
+
```

Comment:
This still seems to be doing things the mutating way, something might have gone awry during the reverting process?

Reply:
Yep, messed something up trying to revert. Removed this, thanks.